setup/sql/dbupdate_04.php

```diff
@@ -20188,7 +20191,9 @@ function writeCtrlClassEntry(ilPluginSlot $slot, array $plugin_data) {
 <?php
 if($ilDB->tableExists('cal_categories_hidden') )
 {
+	$ilDB->dropPrimaryKey('cal_categories_hidden');
```

Comment:
@smeyer-ilias or @alex40724 , this LGTM, how about you?

Reply:
The lines do no harm. But I wonder why they are necessary. Does renameTable only work without primary keys on postgres? This would be different from MySQL, so the DB classes should take care of this imo.

Reply:
Primary Keys in PostgreSQL have an unique name rather than in MySQL, so renaming a table doesn't  rename the constraint.

In step 5168 of dbupdate the primary key of auth_ext_attr_mapping will be dropped, but the constraint-name  is still called saml_attribute_mapping_(pk_idx).

Reply:
ok, maybe i should revert the recreation of primary keys and rename them automatically when renaming the table? what about other constrains and indexes? should they renamed also?

Reply:
If you find the time to investigate how this can be done the best way, I‘d really appreciate since I currently have no running Postgres-based installation running. Thank you very much!